### PR TITLE
Loaded 1669 dynamic pricing on subs page

### PR DESCRIPTION
**Pull request by chris-baxter**

### CODE REVIEW

File: Source/Chargify.NET/Interfaces/IComponentInfo.cs [hunk 1]
```diff
@@ -59,6 +61,32 @@ public enum ComponentType
...
 
+    public class ComponentTypeConverter : JsonConverter<ComponentType>
+    {
+        public override ComponentType Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            string value = reader.GetString().Replace("_", "").ToLowerInvariant();
+
+            switch (value)
+            {
+                case "metered_component":
```
Comment (gus-loaded) [inline Source/Chargify.NET/Interfaces/IComponentInfo.cs:72]: Won't the `.Replace` operation on line 68 mean that this always falls into the `default` branch?

File: Source/Chargify.NET/ChargifyConnect.cs [hunk 2]
```diff
@@ -3741,6 +3742,53 @@ public ICoupon LoadCoupon(int productFamilyId, int couponId)
...
+        /// <param name="ProductFamilyID">The ID of the product family that the coupon belongs to</param>
+        /// <returns>A dictionary of objects if found, empty collection otherwise.</returns>
+        public ICoupon GetCouponByCode(string couponCode, int ProductFamilyID = 0)
+        {
+            try
+            {
+                // make sure data is valid
+                if (ProductFamilyID < 0) throw new ArgumentException("Invalid ProductFamilyID");
+                // now make the request
+                string requestPath = string.Format("coupons/find.{0}?code={1}", GetMethodExtension(), couponCode);
```
Comment (gus-loaded) [inline Source/Chargify.NET/ChargifyConnect.cs]: We should probably URLEncode the coupon code here

File: Source/Chargify.NET/Interfaces/IComponentInfo.cs (hunk 1 above)
Comment (chris-baxter) [reply Source/Chargify.NET/Interfaces/IComponentInfo.cs:72]: Good spotting!

Comment (chris-baxter) [reply Source/Chargify.NET/Interfaces/IComponentInfo.cs:72]: I mean, generally we don't actually care about this value, but to be complete i'll fix it.

**Review by gus-loaded — APPROVED** [review]

**Pull request merged**